starter-project-backend/src/controllers/article.ts

```diff
@@ -0,0 +1,142 @@
+import {Response, Request, NextFunction} from "express"
+import Article  from "../models/article"
+
+async function getAllArticle(req:Request, res:Response, next:NextFunction) 
+{
+  try{
+    let articles = await Article.find({}) 
+    res.status(200).json({data: articles})
+  }catch(e)
+  {
+    res.status(500).send("server error")
+  }
+
+}
+
+async function getArticleById( req:Request,res:Response, next:NextFunction) {
+
+
+  if(!req.params.id) {
+    res.status(400).send("bad request")
+    return;
+  }
+
+
+  try{
+
+    let article = await Article.findOne({id: req.params.id}) 
+
+    if(!article){ 
+      res.status(404).json("article not found")
+      return;
+    }
+    
+    res.status(200).json({data: article})
+
+  }catch(e) {
+    res.status(500).send("data error")
+  }
+
+
+}
+
+async function addArticle(req:Request, res:Response,  next:NextFunction) 
+{
+  //TODO: what if the same object exists already
+  let {id, author, title, content} = req.body
+
```

Comment:
it's not a good idea to pass id from body security-wise, mongoose can generate a random arbitrary _id object